modular_skyrat/modules/modular_implants/code/soulcatcher/soulcatcher_items.dm

```diff
@@ -0,0 +1,143 @@
+		return TRUE
+
+	if(!target_mob.mind || !target_mob.ckey)
+		to_chat(user, span_warning("You are unable to remove a mind from an empty body."))
+		return FALSE
+
+	if(target_mob.stat == DEAD) //We can temporarily store souls of dead mobs.
+		target_mob.ghostize(TRUE) //Incase they are staying in the body.
+		var/mob/dead/observer/target_ghost = target_mob.get_ghost(TRUE, TRUE)
+		if(!target_ghost)
+			to_chat(user, span_warning("You are unable to get the soul of [target_mob]!"))
+			return FALSE
+
+		var/datum/soulcatcher_room/target_room = tgui_input_list(user, "Chose a room to send [target_mob]'s soul to.", name, linked_soulcatcher.soulcatcher_rooms)
+		if(!target_room)
+			return FALSE
+
+		if(tgui_alert(target_ghost, "[user] wants to transfer you to a soulcatcher, do you accept?", name, list("Yes", "No"), autofocus = TRUE) != "Yes")
+			to_chat(user, span_warning("[target_mob] doesn't seem to want to enter."))
+			return FALSE
+
+		if(!target_room.add_soul_from_ghost(target_ghost))
+			return FALSE
+
+		body_component = target_mob.GetComponent(/datum/component/previous_body)
+		if(!body_component)
+			return FALSE
+
+		log_admin("[user] used [src] to put [target_mob]'s mind into a soulcatcher.")
+		scan_body(body_component, user)
+		return TRUE
+
+	var/datum/soulcatcher_room/target_room = tgui_input_list(user, "Chose a room to send [target_mob]'s soul to.", name, linked_soulcatcher.soulcatcher_rooms)
+	if(!target_room)
+		return FALSE
+
+	if(tgui_alert(target_mob, "Do you wish to enter [target_room]?", name, list("Yes", "No")) != "Yes")
+		to_chat(user, span_warning("[target_mob] doesn't seem to want to enter."))
+		return FALSE
+
+	target_room.add_soul(target_mob.mind, TRUE)
+	playsound(src, 'modular_skyrat/modules/modular_implants/sounds/default_good.ogg', 50, FALSE, ignore_walls = FALSE)
+	visible_message(span_notice("[src] beeps: [target_mob]'s mind transfer is now complete."))
+
+	body_component = target_mob.GetComponent(/datum/component/previous_body)
+	scan_body(body_component, user)
+	log_admin("[user] used [src] to put [target_mob]'s mind into a soulcatcher while [target_mob] was still alive.")
+
+	return TRUE
+
+/obj/item/soulcatcher_item/attack_secondary(mob/living/target_mob, mob/living/user, params)
+	if(!target_mob)
+		return FALSE
+
+	if(target_mob.mind || target_mob.ckey || !ishuman(target_mob))
+		to_chat(user, span_warning("[target_mob] is not able to recieve a soul"))
+		return FALSE
+
+	var/list/soul_list = list()
+	for(var/datum/soulcatcher_room/room in linked_soulcatcher.soulcatcher_rooms)
+		for(var/mob/living/soulcatcher_soul/soul in room.current_souls)
+			if(!soul.round_participant || soul.body_scan_needed)
+				continue
+
+			soul_list += soul
+
+	if(length(soul_list) == 0)
+		to_chat(user, span_warning("There are no souls that can be transfered to [target_mob]"))
+		return FALSE
+
+	var/mob/living/soulcatcher_soul/chosen_soul = tgui_input_list(user, "Chose a soul to transfer into the body", name, soul_list)
```

Comment:
Wait wait wait, hold on. You can use this item to mindswap people, so long as they're dead? That feels like a lot more than something that's supposed to be a roleplay tool first and foremost.

Doesn't that also mean you could bring someone that's a ghost, into another body, and thus basically into the round, at basically no cost whatsoever?

Reply:
It's possible to do this yes, but I fail to see how this differs from physically swapping the brains of two people. This also requires the target body to lack a ckey or mind.  Could you describe to me how it is different from that in a meaningful way?

Secondly no. When making the list of souls that could be transferred, any soul that wasn't inside of the round was excluded.

Reply:
also, I will add a check to see if the target body's mind is currently inside of a soulcatcher, if that is a potential issue.